nft/ethereum_payment_obligation.go

```diff
@@ -80,45 +88,155 @@ func newEthereumPaymentObligation(
 	}
 }
 
-func (s *ethereumPaymentObligation) prepareMintRequest(ctx context.Context, tokenID TokenID, documentID []byte, depositAddress string, proofFields []string) (MintRequest, error) {
-	model, err := s.docSrv.GetCurrentVersion(ctx, documentID)
+func (s *ethereumPaymentObligation) prepareMintRequest(ctx context.Context, tokenID TokenID, cid identity.CentID, req MintNFTRequest) (mreq MintRequest, err error) {
+	model, err := s.docSrv.GetCurrentVersion(ctx, req.DocumentID)
 	if err != nil {
-		return MintRequest{}, err
+		return mreq, err
 	}
 
 	corDoc, err := model.PackCoreDocument()
 	if err != nil {
-		return MintRequest{}, err
+		return mreq, err
 	}
 
-	proofs, err := s.docSrv.CreateProofs(ctx, documentID, proofFields)
+	docProofs, err := s.docSrv.CreateProofs(ctx, req.DocumentID, req.ProofFields)
 	if err != nil {
-		return MintRequest{}, err
+		return mreq, err
+	}
+
+	nftProofs, err := generateNFTProofs(corDoc, req, tokenID, cid)
+	if err != nil {
+		return mreq, err
+	}
+
+	var pfs []*proofspb.Proof
+	pfs = append(pfs, docProofs.FieldProofs...)
+	for _, p := range nftProofs {
+		p := p
+		pfs = append(pfs, &p)
 	}
 
-	toAddress := common.HexToAddress(depositAddress)
+	toAddress := common.HexToAddress(req.DepositAddress)
```

Comment:
Wouldn't be nice not to "rename" the address here and call it `depositAddress`?
```suggestion
	depositAddress := common.HexToAddress(req.DepositAddress)
```